### PR TITLE
Jest fixes

### PR DESCRIPTION
**Pull request by johnrees**
https://github.com/wikihouseproject/buildx-editor/commit/2585e7d8173ea94f4c8726ccfe2b6513be3a350b addresses the issue of jest swallowing console.logs and throwing an error

https://github.com/wikihouseproject/buildx-editor/commit/1c6fc7b4c8a855cbd20c95f12c3872bb12707d2d is a WIP, right now jest doesn't finish if there is an error, because there are still some processes running. I think it's either express or jsjob not being shut down correctly. @jonnor I've added `QUESTION:` comments in the code where I think it might be useful to look?

### CODE REVIEW

**Comment by johnrees** [comment]
Can also bind to the server.close() event with something like

```javascript
    const server = app.listen(port, (err) => {
      server.on('close', () => {
        runner.stop()
      })
      return callback(err, server, port)
    })
```

I tried a bunch of things and when I was commenting out `app.post('/nest', doNestingSync)` or putting res.end inside doNestingSync() it was behaving correctly which makes me think it might be that jsjob just needs cleaning up?

**Comment by jonnor** [comment]
Yes `JsJob.Runner` should be cleaned up when server is.

**Comment by jonnor** [comment]
Note that `http.Server 'close'` event has the problem (at least did in Node 4.x) that in case of in-flight requests it wont get fired until they are done. Usually not a problem, but can happen when test fails due to timeout.

File: __tests__/SVGnest/index.js [hunk 1]
```diff
@@ -37,18 +37,20 @@ describe('Nesting', () => {
...
 
     it('should return cutsheets', function (done) {
+
       const req = superagent
         .post(`http://localhost:${serverOptions.port}/nest`)
         .field('bin', 'rect5794')
         .attach('svg', svgData, 'svg')
-      req.end((err, res) => {
+
+      req.end(catchErrors(done, (err, res) => {
```
Comment (jonnor) [inline __tests__/SVGnest/index.js]: An alternative to this would be to use Promises instead of call-continuation callbacks. Might be cleaner?

**Comment by jonnor** [comment]
Found and fixed an issue in JsJob, https://github.com/flowhub/jsjob/commit/c01c48885b9e634609aad92aa07be5f5c4bff246

The module [wtfnode](https://www.npmjs.com/package/wtfnode) is useful for debugging node.js event loop not completing, it tells what things are still pending.

**Comment by jonnor** [comment]
@johnrees `npm test` should now exit correctly on completion

**Comment by johnrees** [comment]
Nice! haha @ wtfnode

**Pull request merged**